File: CASE_Simple_Example_V4/Hardened/HAMR/src/main/component/HAMR/SW/UARTDriver_Impl_SW_UART_UARTDriver.scala
// #Sireum

package HAMR.SW

import org.sireum._
import HAMR._

// This file will not be overwritten so is safe to edit
object UARTDriver_Impl_SW_UART_UARTDriver {

  def initialise(api: UARTDriver_Impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.put_Status(SW.Coordinate_Impl.example())
  }

  def handle_MissionWindow(api: UARTDriver_Impl_Operational_Api, value : SW.MissionWindow): Unit = {
    api.logInfo("example handle_MissionWindow implementation")
    api.logInfo(s"received ${value}")
    // example api usage

    val apiUsage_MissionWindow: Option[SW.MissionWindow] = api.get_MissionWindow()
    api.logInfo(s"Received on MissionWindow: ${apiUsage_MissionWindow}")
  }

  def activate(api: UARTDriver_Impl_Operational_Api): Unit = { }

  def deactivate(api: UARTDriver_Impl_Operational_Api): Unit = { }

  def finalise(api: UARTDriver_Impl_Operational_Api): Unit = { }

  def recover(api: UARTDriver_Impl_Operational_Api): Unit = { }
}

File: CASE_Simple_Example_V4/Hardened/HAMR/hamr/ext-c/UARTDriver_Impl_SW_UART_UARTDriver/UARTDriver_Impl_SW_UART_UARTDriver.c
#include <UARTDriver_Impl_SW_UART_UARTDriver_api.h>
#include <UARTDriver_Impl_SW_UART_UARTDriver.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "UARTDriver_Impl_SW_UART_UARTDriver.c", "", "HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_initialise_", 0);

  // examples of api setter and logging usage

  DeclNewHAMR_SW_Coordinate_Impl(t0);
  HAMR_SW_Coordinate_Impl_example(SF &t0);
  api_put_Status__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF &t0);

  api_logInfo__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF string("Example logInfo"));

  api_logDebug__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF string("Example logDebug"));

  api_logError__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF string("Example logError"));
}

Unit HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "UARTDriver_Impl_SW_UART_UARTDriver.c", "", "HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_finalise_", 0);
}

Unit HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_handle_MissionWindow_(
  STACK_FRAME
  HAMR_SW_MissionWindow value) {
  DeclNewStackFrame(caller, "UARTDriver_Impl_SW_UART_UARTDriver.c", "", "HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_handle_MissionWindow_", 0);

  DeclNewString(MissionWindowString);
  String__append(SF (String) &MissionWindowString, string("HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver_handle_MissionWindow called"));
  api_logInfo__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver (SF (String) &MissionWindowString);

  DeclNewString(_str);
  String__append(SF (String) &_str, string("Received on MissionWindow: "));
  HAMR_SW_MissionWindow_string_(SF (String) &_str, value);
  api_logInfo__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF (String) &_str);

  // examples of api getter usage

  DeclNewHAMR_SW_MissionWindow(t0);
  if(api_get_MissionWindow__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF &t0)) {
    DeclNewString(MissionWindow_str);
    String__append(SF (String) &MissionWindow_str, string("Received on MissionWindow: "));
    HAMR_SW_MissionWindow_string_(SF (String) &MissionWindow_str, &t0);
    api_logInfo__HAMR_SW_UARTDriver_Impl_SW_UART_UARTDriver(SF (String) &MissionWindow_str);
  }
}
